client/block.c

```diff
@@ -70,13 +76,26 @@ struct block_backrefs {
 #define ourprev link[MAX_LINKS - 2]
 #define ournext link[MAX_LINKS - 1]
 
+struct cache_block {
+        struct ldus_rbtree node;
+	xdag_hash_t hash;
+        struct xdag_block block;
+        struct cache_block *next;
+};
+
+
 static xdag_amount_t g_balance = 0;
 static xdag_time_t time_limit = DEF_TIME_LIMIT, xdag_era = XDAG_MAIN_ERA;
-static struct ldus_rbtree *root = 0;
+static struct ldus_rbtree *root = 0, *cache_root = 0;
 static struct block_internal *volatile top_main_chain = 0, *volatile pretop_main_chain = 0;
 static struct block_internal *ourfirst = 0, *ourlast = 0, *noref_first = 0, *noref_last = 0;
+static struct cache_block *cache_first = NULL, *cache_last = NULL;
 static pthread_mutex_t block_mutex;
 static int g_light_mode = 0;
+static uint32_t cache_bounded_counter = 0;
+
+// functions
+void xdag_print_block_list(struct block_internal**, int, int, FILE*);
```

Comment:
Better to remove that declaration. This function was removed in https://github.com/XDagger/xdag/pull/215

Reply:
cannot compile without it